miniapp/include/dlaf/miniapp/options.h

```diff
@@ -0,0 +1,264 @@
+  MiniappOptions& operator=(MiniappOptions&&) = default;
+  MiniappOptions& operator=(const MiniappOptions&) = default;
+};
+
+inline hpx::program_options::options_description getMiniappOptionsDescription() {
+  hpx::program_options::options_description desc("DLA-Future miniapp options");
+
+  desc.add_options()(
+      "backend", hpx::program_options::value<std::string>()->default_value("default"),
+      "Backend to use ('default' ('gpu' if available, otherwise 'mc'), 'mc', 'gpu' (if available))");
+  desc.add_options()(
+      "type", hpx::program_options::value<std::string>()->default_value("d"),
+      "Element type to use ('s' (float), 'd' (double), 'c' (std::complex<single>), 'z' (std::complex<double>))");
+  desc.add_options()("grid-rows", hpx::program_options::value<int>()->default_value(1),
+                     "Number of row processes in the 2D communicator");
+  desc.add_options()("grid-cols", hpx::program_options::value<int>()->default_value(1),
+                     "Number of column processes in the 2D communicator");
+  desc.add_options()("nruns", hpx::program_options::value<int64_t>()->default_value(1),
+                     "Number of runs");
+  desc.add_options()("nwarmups", hpx::program_options::value<int64_t>()->default_value(1),
+                     "Number of warmup runs");
+  desc.add_options()("check-result", hpx::program_options::value<std::string>()->default_value("none"),
+                     "Enable result checking ('none', 'all', 'last')");
+
+  return desc;
+}
+
+inline void addLayoutOption(hpx::program_options::options_description& desc,
+                            const std::string& def = "C") {
+  desc.add_options()("layout", hpx::program_options::value<std::string>()->default_value(def),
+                     "'C' (ColMajor), 'R' (RowMajor)");
+}
+
+inline void addOpOption(hpx::program_options::options_description& desc, const std::string& def = "N") {
+  desc.add_options()("op", hpx::program_options::value<std::string>()->default_value(def),
+                     "'N' (NoTrans), 'T' (Trans), 'C' (ConjTrans)");
+}
+
+inline void addUploOption(hpx::program_options::options_description& desc,
+                          const std::string& def = "L") {
+  desc.add_options()("uplo", hpx::program_options::value<std::string>()->default_value(def),
+                     "'L' (Lower), 'U' (Upper), 'G' (General)");
+}
+
+inline void addDiagOption(hpx::program_options::options_description& desc,
+                          const std::string& def = "N") {
+  desc.add_options()("diag", hpx::program_options::value<std::string>()->default_value(def),
+                     "'N' (NonUnit), 'U' (Unit)");
+}
+
+inline void addSideOption(hpx::program_options::options_description& desc,
+                          const std::string& def = "L") {
+  desc.add_options()("side", hpx::program_options::value<std::string>()->default_value(def),
+                     "'L' (Left), 'R' (Right)");
+}
```

Comment:
Yeah, I think your suggestion makes sense. I was a bit unsure how much to reuse the `blaspp` functionality.

Reply:
In general I rely on blaspp functionalities because their enums are part of our public interface, so we explicitly rely on them at the moment...and so I fully rely also on the utils 😄

Reply:
This is now using `side2char` and friends (note, not `side2str`). This is to match with the parsing functions which emulate `char2side` and friends (see https://github.com/eth-cscs/DLA-Future/pull/443#issuecomment-1006568214). Does this look ok to you?